File: Examples/CSharp/Working-with-Print/SetPrintJobAndPrinterName.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_with_Print
{
    public class SetPrintJobAndPrinterName
    {
        public static void Run() 
        {
            // ExStart:SetPrintJobAndPrinterName
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Print();

            // load source Visio diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");

            // call the print method to print whole Diagram using the printer name and set document name in the print job
            diagram.Print("LaserJet1100", "Job name while printing with Aspose.Diagram");
            // ExEnd:SetPrintJobAndPrinterName
        }
    }
}

File: Examples/VisualBasic/Working-with-Print/SetPrintJobAndPrinterName.vb
Imports Aspose.Diagram
Imports System
Public Class SetPrintJobAndPrinterName
    Public Shared Sub Run()
        ' ExStart:SetPrintJobAndPrinterName
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Print()

        ' Load source Visio diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))

        ' Call the print method to print whole Diagram using the printer name and set document name in the print job
        diagram.Print("LaserJet1100", "Job name while printing with Aspose.Diagram")
        ' ExEnd:SetPrintJobAndPrinterName
    End Sub
End Class
